### PR TITLE
Add tools for querying bug status in Bugzilla

### PR DESCRIPTION
**Pull request by mdoucha**
Some bugs take a while to fix which creates a dilemma: mark them as softfail and risk letting broken fix slip through testing, or let them keep failing and increase the workload of test reviewers.

This patchset adds tools for querying bug status in Bugzilla which can be used to automatically disable softfails for known issues once the corresponding bug is closed and ensure full testing of the fix. LTP known issues filter is included as example implementation.

- Related ticket: N/A
- Needles: N/A
- Verification run:
  - SLE-12SP5 ltp_syscalls: https://openqa.suse.de/tests/6290406 (ptrace10 expected to fail due to bugzilla query error - missing `BUGZILLA_URL`)
  - SLE-12SP5 ltp_syscalls: https://openqa.suse.de/tests/6290407 (ptrace10 should softfail)
  - SLE-15SP2 ltp_kernel_misc: https://openqa.suse.de/tests/6290408 (zram01-03 should softfail)

### CODE REVIEW

**Comment by mdoucha** [comment]
All verification runs finished with the expected result. This PR is ready for review and merge.

File: lib/bugzilla.pm
Comment (okurz) [inline lib/bugzilla.pm:84]: Please avoid the use of bmwqemu outside of os-autoinst. That module should not be considered outside facing API. The test API offers`diag` directly

Comment (mdoucha) [reply lib/bugzilla.pm]: Fixed, verification runs restarted.

**Review by asmorodskyi — APPROVED** [review]
nice !

File: lib/bugzilla.pm
Comment (b10n1k) [inline lib/bugzilla.pm:19]: i dont think you need this as you use the `Exporter` as base

File: lib/bugzilla.pm
Comment (b10n1k) [inline lib/bugzilla.pm:35]: nitpicking: could you remove whitespace?

**Review by b10n1k — CHANGES_REQUESTED** [review]
i have one question? Would it make sense to set `BUGZILLA_URL` in main? this value should not change a lot along the tests? wdyt?

**Review by czerw — APPROVED** [review]
Nice!

**Comment by mdoucha** [comment]
> i have one question? Would it make sense to set `BUGZILLA_URL` in main? this value should not change a lot along the tests. wdyt?

We'll have to point `BUGZILLA_URL` to our Fastzilla instance because regular Bugzilla does not allow access to some bugs. Putting `BUGZILLA_URL` in main would cause trouble to people outside SUSE who run these tests on their own.

**Comment by b10n1k** [comment]
> > i have one question? Would it make sense to set `BUGZILLA_URL` in main? this value should not change a lot along the tests. wdyt?
> 
> We'll have to point `BUGZILLA_URL` to our Fastzilla instance because regular Bugzilla does not allow access to some bugs. Putting `BUGZILLA_URL` in main would cause trouble to people outside SUSE who run these tests on their own.


> > i have one question? Would it make sense to set `BUGZILLA_URL` in main? this value should not change a lot along the tests. wdyt?
> 
> We'll have to point `BUGZILLA_URL` to our Fastzilla instance because regular Bugzilla does not allow access to some bugs. Putting `BUGZILLA_URL` in main would cause trouble to people outside SUSE who run these tests on their own.

sorry if i was not clear. by main i meant the `main_common.pm` of this repo.

**Comment by mdoucha** [comment]
> > We'll have to point `BUGZILLA_URL` to our Fastzilla instance because regular Bugzilla does not allow access to some bugs. Putting `BUGZILLA_URL` in main would cause trouble to people outside SUSE who run these tests on their own.
> 
> sorry if i was not clear. by main i meant the `main_common.pm` of this repo.

Yes, that's how I understood it from the beginning. I still think that putting the URL of an internal company service into public code is a terrible idea, in part due to security concerns.

**Comment by okurz** [comment]
I see no problem in providing a URL that does not leak any secrets as "sane default" unless otherwise configured

**Comment by mdoucha** [comment]
> I see no problem in providing a URL that does not leak any secrets as "sane default" unless otherwise configured

It would not be a sane default for anyone outside the company. The code handles missing `BUGZILLA_URL` by returning `undef` immediately, which should be treated as generic query error by the caller. If we hardcode a default URL, we'll slow down testing for people outside SUSE because now the code will have to wait for DNS query failure.

`BUGZILLA_URL` should be set in the medium or as a global variable on the OpenQA instance. Not in the code.

**Review by b10n1k — APPROVED** [review]
+1

**Comment by okurz** [comment]
please rebase to fix the conflict. After that please merge yourself, I approve

**Comment by mdoucha** [comment]
Rebased and retested. But I will not merge this myself because:
1) I'm strongly against anyone merging their own code because it defeats the purpose of code review.
2) I don't have merge privileges in this repo.

**Comment by okurz** [comment]
>     1. I'm strongly against anyone merging their own code because it defeats the purpose of code review.

The purpose of code review is to have code reviewed by reviewers. That has clearly happened here already so I don't see the problem. The benefit of merging yourself especially for this repo is that you can choose a time suitable for yourself when afterwards you can monitor for any bad impact, i.e. tests failing due to unexpected side-effects from yours changes.

**Comment by okurz** [comment]
I invited you to the organisation and the team of test maintainers.

**Pull request merged**
